logstash-core/lib/logstash/instrument/metric.rb

```diff
@@ -8,87 +8,102 @@ class MetricNoKeyProvided < MetricException; end
   class MetricNoBlockProvided < MetricException; end
   class MetricNoNamespaceProvided < MetricException; end
 
+  # This class provide the interface between the code, the collector and the format
+  # of the recorded metric.
   class Metric
-    class TimedExecution
-      MILLISECONDS = 1_000_000.0.freeze
-
-      def initialize(metric, key)
-        @metric = metric
-        @key = key
-        start
-      end
-
-      def start
-        @start_time = current_time
-      end
-
-      def stop
-        @metric.report_time(@key, (MILLISECONDS * (current_time - @start_time)).to_i)
-      end
-
-      private
-      def current_time
-        Time.now
-      end
-    end
-
-    attr_reader :collector, :namespace_information
+    attr_reader :collector
 
-    public
-    def initialize(collector, namespace = nil)
+    def initialize(collector)
       @collector = collector
-      @namespace_information = Array(namespace).map(&:to_sym)
     end
 
-    def increment(key, value = 1)
+    def increment(namespace, key, value = 1)
       validate_key!(key)
-      collector.push(namespace_information, key, :counter, :increment, value)
+      collector.push(namespace, key, :counter, :increment, value)
```

Comment:
Yes, but this is intended for custom internal pipeline. Lets say you want to update 2 different keys in the same code context.

```
stats.pipelines.events_in
stats.pipelines.pipelines01.plugins.XXXXX.events_in
```

This require 2 different namespace, I think is an OK trade off, we protect the plugins author to create metrics anywhere they want, but we are still able to do it.